Assets/Scripts/Effects/GenerateWaterPlane.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace Effects
+{
+    [ExecuteInEditMode]
+    public class GenerateWaterPlane : MonoBehaviour
+    {
+        [SerializeField][Tooltip("Size of the mesh")]
+        private float size = 1f;
+    
+        [SerializeField][Tooltip("Subdivisions of the mesh")]
+        private int gridSize = 16;
+
+        private MeshFilter _filter;
+
+        // Start is called before the first frame update
+        void Start()
+        {
+            _filter = GetComponent<MeshFilter>();
+            _filter.mesh = generateMesh();
+        }
+
+        private Mesh generateMesh()
```

Comment:
Methods should start with a capital letter. Please also add summaries with /// for the methods.

Reply:
Don't we have summaries for public methods only?
(Will change method name)

Reply:
ye no, not specified xD I'm a fan of summaries ;) so everyone knows what's going on. But it's up to you, I guess.